firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -49,37 +49,60 @@ void disableMotors()
 // Drives the motor corresponding to "motor" to it's home position (position 0)
 void homeAxis(int motor[])
 {
+  int * currentMotorPos;
+  int currentMotorDir, currentMotorNumEighthSteps;
   int i;
 
+  //  Stores corresponding motor position based off of which motor is being homed,
+  // so that correct position can be incremented by function
+  currentMotorPos = (motor == xMotor) ? &currentX : &currentY;
+
   // Loop until endstop collision, then fine tune it
   // LEFT and DOWN have same value, as do RIGHT and UP. Use LEFT
   // and RIGHT arbitrarily while tuning end stop for both x and y axes.
 
+  // Sets scale and direction for motor and current position
+  // Moving motor towards home for rough estimate
   digitalWrite(motor[DIR_PIN], LEFT);
   setScale(motor, WHOLE_STEPS);
+  currentMotorNumEighthSteps = 8;
+  currentMotorNumEighthSteps *= -1;
   while (digitalRead(motor[ENDSTOP_PIN]) == LOW)
   {
     digitalWrite(motor[STEP_PIN], LOW);
     delay(1);
     digitalWrite(motor[STEP_PIN], HIGH);
+
+    // Motor is moved above, current position for whatever motor updated here
+    *currentMotorPos += currentMotorNumEighthSteps;
   }
 
+  // Flipping direction again to move motor away from home to prepare for fine-tuning
```

Comment:
In general, try to have comments that start with a verb `Does blah` instead of `Doing blah`. So in this case it would be `Flips direction ...`